Sources/AppcuesKit/Push/PushEnvironment.swift

```diff
@@ -0,0 +1,144 @@
+    enum PushEnvironmentError: LocalizedError {
+        case noEntitlementsKey
+        case noEnvironmentKey
+        case unexpectedEnvironment(String)
+
+        var errorDescription: String? {
+            switch self {
+            case .noEntitlementsKey: return "No 'Entitlements' key"
+            case .noEnvironmentKey: return "No entitlement 'aps-environment' key"
+            case .unexpectedEnvironment(let value): return "Unexpected 'aps-environment' value '\(value)'"
+            }
+        }
+    }
+
+    enum Reason {
+        case notComputed, error(Error)
+
+        var description: String {
+            switch self {
+            case .notComputed: return "Value not computed"
+            case .error(let error): return error.localizedDescription
+            }
+        }
+    }
+
+    var environmentValue: String {
+        // The environment to request from the backend must be "development" or "production".
+        // If we haven't been able to determine the environment, default to "production".
+        switch self {
+        case .development: return "development"
+        case .unknown, .production: return "production"
+        }
+    }
+
+    init?(value: String) {
+        switch value {
+        case "development": self = .development
+        case "production": self = .production
+        default: return nil
+        }
+    }
+}
+
+extension UIDevice {
+    enum ProvisioningProfileError: LocalizedError {
+        case noEmbeddedProfile
+        case plistScanFailed
+        case plistSerializationFailed
+
+        var errorDescription: String? {
+            switch self {
+            case .noEmbeddedProfile: return "No 'embedded.mobileprovision' found in bundle"
+            case .plistScanFailed: return "Property list scan failed"
+            case .plistSerializationFailed: return "Property list serialization failed"
+            }
+        }
+    }
+
+    func pushEnvironment() -> PushEnvironment {
+    #if targetEnvironment(simulator)
+    return .development
+    #else
+    do {
+        let provisioningProfile = try UIDevice.current.provisioningProfile()
+
+        guard let entitlements = provisioningProfile["Entitlements"] as? [String: Any] else {
+            return .unknown(.error(PushEnvironment.PushEnvironmentError.noEntitlementsKey))
+        }
+
+        guard let environment = entitlements["aps-environment"] as? String else {
+            return .unknown(.error(PushEnvironment.PushEnvironmentError.noEnvironmentKey))
+        }
+
+        guard let pushEnvironment = PushEnvironment(value: environment) else {
+            return .unknown(.error(PushEnvironment.PushEnvironmentError.unexpectedEnvironment(environment)))
+        }
+
+        return pushEnvironment
+    } catch ProvisioningProfileError.noEmbeddedProfile {
+        // App Store apps do not contain an embedded provisioning profile,
+        // and since we know we're not on a simulator, that means it's "production".
+        return .production
```

Comment:
This is the important part to not show a debugger error for App Store builds ([slack ref](https://appcues.slack.com/archives/C031RLGS4F8/p1716391502737889?thread_ts=1716296544.624109&cid=C031RLGS4F8))